### PR TITLE
Loan RPC: `withdrawFromVault`

### PR DESCRIPTION
**Pull request by ivan-zynesis**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Mandatory rpc for loan implementation.

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

**Comment by ivan-zynesis** [comment]
Awaiting cpp team response on failed test RPC behavior https://cakedefi.slack.com/archives/G01LSAS1KRQ/p1635824643007800 , most likely a bug need to be fixed.

File: packages/jellyfish-api-core/__tests__/category/loan/withdrawFromVault.test.ts [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+      const utxo = await tGroup.get(0).container.fundAddress(vaultOwner, 10)
+      const txid = await tGroup.get(0).rpc.loan.withdrawFromVault({
+        vaultId: vaultId1,
+        to: collateralAddress,
+        amount: '2.34567@DFI'
+      }, [utxo])
+
+      const rawtx = await tGroup.get(0).container.call('getrawtransaction', [txid, true])
+      expect(rawtx.vin[0].txid).toStrictEqual(utxo.txid)
+      expect(rawtx.vin[0].vout).toStrictEqual(utxo.vout)
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/withdrawFromVault.test.ts:260]: maybe add a line to check `collateralAddress` received the withdrawn amount for completeness.

File: packages/jellyfish-api-core/__tests__/category/loan/withdrawFromVault.test.ts [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+      {
+        // reduce DFI value to below 50%
+        await tGroup.get(0).rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '0.4999@DFI', currency: 'USD' }] })
+        await tGroup.get(0).generate(12)
+      }
+
+      const promise = tGroup.get(0).rpc.loan.withdrawFromVault({
+        vaultId: vaultId3,
+        to: await tGroup.get(0).generateAddress(),
+        amount: '1@DFI' // $4000 of DFI vs $5000 of BTC (tested with 5001, do not work, there are extra margin)
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/withdrawFromVault.test.ts:318]: the comment needs to be updated?

**Review by surangap — COMMENTED** [review]
looks good. minor comments

**Review by surangap — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**